processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
@gianm IMO it makes more sense to emit the duration for the "primary" data source of the join query as `duration` dimension of a query metric. Also we could add dimension like "secondDataSourceDuration".

Also to add to your message, now with `QueryMetrics` we should lean more towards not emitting metrics and rather add no-op hooks, so that people who need specific dimensions/metrics could make it in extensions.